File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const cnt = {};
    for (const person of participant) {
        if (person in cnt) cnt[person] += 1;
        else cnt[person] = 1;
    }
    for (const person of completion) {
        cnt[person] -= 1;
    }
    for (const person in cnt) {
        if (cnt[person] !== 0) return person
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> people = new HashMap<>();
        
        for (String startPerson : participant) {
            people.put(startPerson, people.getOrDefault(startPerson, 0) + 1);
        }

        for (String completedPerson : completion) {
            people.put(completedPerson, people.get(completedPerson) - 1);
        }
        
        for (String person : people.keySet()) {
            if (people.get(person) > 0) {
                return person;
            }
        }
        
        return "";
    }
}